### PR TITLE
Initial version of the box tables

### LINKED ISSUE #167: AVIF should have a table of boxes and box versions
Cyril has created a prototype table of required boxes/box-versions for stills and image sequences which would be a good addition to the spec. This issue will track getting that added at some point.

### PR DESCRIPTION
**Pull request by cconcolato**
closes #167

***
<a href="https://pr-preview.s3.amazonaws.com/AOMediaCodec/av1-avif/pull/170.html" title="Last updated on Dec 8, 2021, 4:59 PM UTC (8a0ce79)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/AOMediaCodec/av1-avif/170/a1e8bc1...8a0ce79.html" title="Last updated on Dec 8, 2021, 4:59 PM UTC (8a0ce79)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -446,6 +446,359 @@ A file containing a 'pict' track compliant with this profile is expected to list
...
+            <td>&nbsp;</td>
+            <td>-</td>
+            <td>ISOBMFF</td>
+            <td>meta</td>
+            <td>Used to indicate that multiple images are semantically grouped</td>
+        </tr>
+        <tr>
+            <td>&nbsp;</td>
+            <td>altr</td>
+            <td>-</td>
```
Comment (leo-barnes) [inline index.bs:694]: ISOBMFF lists `altr` as being a full box. This should probably be `0`.

File: index.bs [hunk 2]
```diff
@@ -446,6 +446,359 @@ A file containing a 'pict' track compliant with this profile is expected to list
...
+            <td>&nbsp;</td>
+            <td>0</td>
+            <td>HEIF</td>
+            <td>ipco</td>
+            <td>Used to signal the type of an auxiliary image (e.g. alpha, depth).</td>
+        </tr>
+        <tr>
+            <td>clap</td>
+            <td>&nbsp;</td>
+            <td>0</td>
```
Comment (leo-barnes) [inline index.bs:726]: ISOBMFF lists `clap` as being a simple box. This should probably be `-`.

File: index.bs [hunk 3]
```diff
@@ -446,6 +446,359 @@ A file containing a 'pict' track compliant with this profile is expected to list
...
+            <td>-</td>
+            <td>HEIF</td>
+            <td>ipco</td>
+            <td>Used to signal a mirroring applied to an image</td>
+        </tr>
+        <tr>
+            <td>clli</td>
+            <td>&nbsp;</td>
+            <td>-</td>
+            <td>MIAF</td>
```
Comment (leo-barnes) [inline index.bs:751]: HEIF (`wxxxxx_23008-12_Ed2_FDIS+COR1_R1+FDAM1_R1` specifically) says `clli` is defined in ISOBMFF, but my version (`m47256_14496-12_2015(E)-6th+FDAM1+FDAM2+COR1+KB-R6+6thCOR1+AMD1_draft04-must`) does not have it. Either the HEIF draft is wrong or this should say ISOBMFF instead.

File: index.bs [hunk 4]
```diff
@@ -446,6 +446,359 @@ A file containing a 'pict' track compliant with this profile is expected to list
...
+            <td>-</td>
+            <td>ISOBMFF</td>
+            <td>ipco</td>
+            <td>Used to signal HDR color volume for an image</td>
+        </tr>
+        <tr>
+            <td>mdcv</td>
+            <td>&nbsp;</td>
+            <td>-</td>
+            <td>MIAF</td>
```
Comment (leo-barnes) [inline index.bs:767]: HEIF (`wxxxxx_23008-12_Ed2_FDIS+COR1_R1+FDAM1_R1` specifically) says `mdcv` is defined in ISOBMFF, but my version (`m47256_14496-12_2015(E)-6th+FDAM1+FDAM2+COR1+KB-R6+6thCOR1+AMD1_draft04-must`) does not have it. Either the HEIF draft is wrong or this should say ISOBMFF instead.

(Haven't been able to check if it's a full or simple box due to not finding it)

File: index.bs (hunk 3 above)
Comment (leo-barnes) [reply index.bs:751]: (Haven't been able to check if it's a full or simple box due to not finding it)

File: index.bs [hunk 5]
```diff
@@ -446,6 +446,359 @@ A file containing a 'pict' track compliant with this profile is expected to list
...
+            <td>ipco</td>
+            <td>Used to signal HDR light level information for an image</td>
+        </tr>
+        <tr>
+            <td>cclv</td>
+            <td>&nbsp;</td>
+            <td>-</td>
```
Comment (leo-barnes) [inline index.bs:758]: (Haven't been able to check if it's a full or simple box due to not finding it)

**Review by leo-barnes — COMMENTED** [review]
Looks good! Some inline comments.

File: index.bs (hunk 4 above)
Comment (cconcolato) [reply index.bs:767]: It is now defined in ISOBMFF 7th edition (WG03N0148_19980) and it's a simple box.

File: index.bs (hunk 5 above)
Comment (cconcolato) [reply index.bs:758]: It is defined in ISOBMFF 7th edition (WG03N0148_19980) and it's a simple box.

File: index.bs (hunk 1 above)
Comment (cconcolato) [reply index.bs:694]: Good catch. This is related to this issue https://github.com/MPEGGroup/FileFormat/issues/44

File: index.bs (hunk 2 above)
Comment (cconcolato) [reply index.bs]: Fixed

File: index.bs (hunk 1 above)
Comment (cconcolato) [reply index.bs]: Fixed

**Review by y-guyon — APPROVED** [review]

**Comment by leo-barnes** [comment]
This is superseded by #225 and can be closed once that is merged.

**Comment by leo-barnes** [comment]
Closing this as #225 has been merged.

**Pull request closed without merging**